### ISSUE TITLE
UnlockPart freezing game to Mission Control view. Unable to Quit to Menu

### ISSUE THREAD

**Issue by inigmatus** [jrossignol/ContractConfigurator#371]
Was looking to explore work around to using the ExperimentalPart in providing new career players with basic parts. Tried UnlockPart behavior instead.

Unlocking part and tech freezes game to Mission Control view. Unable to Quit to Menu.

Started with a new Career. No techs researched yet. Created contract with:
```
BEHAVIOUR
	{
		name = UnlockPart
		type = UnlockPart
		
		part = batteryPack
		
		unlockTech = true
	}
```
Accepting contract freezes the game to the Mission Control view.

Debug shows:
```
[Exception]: NullReferenceException: Object reference not set to an instance of an object
```

Here is the full contract to test with:

```
CONTRACT_TYPE
{
	name = SSI-Give-batteryPack
	title = SSI: A free battery pack!
	group = SSI Aerospace
	agent = SSI Aerospace
	
	description = Here is a free battery pack
	
	synopsis = a free battery pack
	
	notes = a free battery pack
	
	completedMessage = here you go.

	minExpiry = 7.0
	maxExpiry = 7.0
	deadline = 0
	cancellable = true
	declinable = true
	autoAccept = false
	prestige = Trivial
   	targetBody = Kerbin
   	maxCompletions = 0
   	maxSimultaneous = 1

 	rewardScience = 0
  	rewardReputation = 0
  	rewardFunds = 0
 	failureReputation = 0
  	failureFunds = 0
  	advanceFunds = 0
	weight = 10.0


	BEHAVIOUR
	{
		name = UnlockPart
		type = UnlockPart
		
		part = batteryPack
		
		unlockTech = true
	}

	
	PARAMETER
	{
		name = Timer
		type = Timer
		
		duration = 5s
		timerType = CONTRACT_ACCEPTANCE
		failContract = false
		
	}

}
```

**Comment by jrossignol** [comment]
Need the full stack trace from that (available in the log file, but not in
the debug window).

**Comment by inigmatus** [comment]
where do i find the log file?

**Comment by jrossignol** [comment]
There's a forum thread with that info...  (haha, just kidding, forums are
teh suck).

Look at KSP. log in the install directory of KSP. On mobile right now,
otherwise I'd give you typical paths for windows (somewhere in program
files (x86))

**Comment by inigmatus** [comment]
5MB log. I see a lot of duplicates as it freezes. Must in some sort of loop.

http://www.mediafire.com/download/k161392gpjv1p9d/KSP.log

**Comment by jrossignol** [comment]
First - omg, you're running full verbose logging on Contract Configurator.  That will potentially slow things down a lot (even I only run at DEBUG, and VERBOSE on a couple specific classes as needed).

Anyway, this is what I needed:
```
[EXC 17:47:51.812] NullReferenceException: Object reference not set to an instance of an object
	ProtoTechNode.Save (.ConfigNode node)
	ResearchAndDevelopment.OnSave (.ConfigNode node)
	ScenarioModule.Save (.ConfigNode node)
	ProtoScenarioModule..ctor (.ScenarioModule module)
	ScenarioRunner.GetUpdatedProtoModules ()
	Game.Updated ()
	GamePersistence.SaveGame (System.String saveFileName, System.String saveFolder, SaveMode saveMode)
	MCSceneSpawner.onMCDespawn ()
	EventVoid.Fire ()
	MCDespawner.BtnExit (.POINTER_INFO& ptr)
	UIButton.OnInput (.POINTER_INFO& ptr)
	AutoSpriteControlBase.OnInput (POINTER_INFO ptr)
	UIManager.DispatchHelper (.POINTER_INFO& curPtr, Int32 camIndex)
	UIManager.DispatchInput ()
	UIManager.Update ()
	UIManager.DidAnyPointerHitUI ()
	SpaceCenterCamera2.InputCamera ()
	SpaceCenterCamera2.Update ()
``
Unfortunately, that's all stock stuff, but given you've reproduced in a fresh save I should be able to do so myself with the contract you've given.  Looking into it now.

**Comment by jrossignol** [comment]
Fixed for 1.8.4.  The problem was that you didn't have part unlocking turn on for the game you were playing.  Now it'll work (although it'll only do something in that case if unlockTech=True.
